server/src/com/cloud/user/AccountManagerImpl.java

```diff
@@ -1734,6 +1734,8 @@ public Boolean doInTransaction(TransactionStatus status) {
                 UserVO newUser = new UserVO(user);
```

Comment:
Why creating a new user entry? I mean, isn't this a matter of simply updating the accountId of the user? Then, you would not change the UUID of the "old" user entry, and you would also not need to update the keys and then mark the user as removed.

Reply:
that would be a nice extra feature ;)

Reply:
Ok. Then, there is one thing I still do not understand. If we want to keep the history, why are we changing the old user UUID? Shouldn't it be the opposite?

Reply:
We are not changing db id which is used for foreign key relations. and we are copying the uuid to the external identity for future reference. through the new user we can then also find any 'real external' ids.

Reply:
Got it. Thanks for the details.